services/core-web/common/actionCreators/permitActionCreator.ts

```diff
@@ -0,0 +1,500 @@
+
+export const updatePermit =
+  (
+    mineGuid: string,
+    permitGuid: string,
+    payload: IPermit
+  ): AppThunk<Promise<AxiosResponse<IPermit>>> =>
+  (dispatch): Promise<AxiosResponse<IPermit>> => {
+    dispatch(request(reducerTypes.UPDATE_PERMIT));
+    dispatch(showLoading());
+    return CustomAxios()
+      .put(
+        `${ENVIRONMENT.apiUrl}${API.PERMITS(mineGuid)}/${permitGuid}`,
+        payload,
+        createRequestHeader()
+      )
+      .then((response: AxiosResponse<IPermit>) => {
+        notification.success({
+          message: `Successfully updated permit`,
+          duration: 10,
+        });
+        dispatch(success(reducerTypes.UPDATE_PERMIT));
+        return response;
+      })
+      .catch((err) => {
+        dispatch(error(reducerTypes.UPDATE_PERMIT));
+        throw new Error(err);
+      })
+      .finally(() => dispatch(hideLoading()));
+  };
+
+export const createPermitAmendment =
+  (
+    mineGuid: string,
+    permitGuid: string,
+    payload: Partial<ICreatePermitAmendmentPayload>
+  ): AppThunk<Promise<AxiosResponse<IPermitAmendment>>> =>
+  (dispatch): Promise<AxiosResponse<IPermitAmendment>> => {
+    dispatch(request(reducerTypes.CREATE_PERMIT_AMENDMENT));
+    dispatch(showLoading("modal"));
+    return CustomAxios()
+      .post(
+        `${ENVIRONMENT.apiUrl}${API.PERMIT_AMENDMENTS(mineGuid, permitGuid)}`,
+        payload,
+        createRequestHeader()
+      )
+      .then((response: AxiosResponse<IPermitAmendment>) => {
+        notification.success({
+          message: "Successfully created a new amendment",
+          duration: 10,
+        });
+        dispatch(success(reducerTypes.CREATE_PERMIT_AMENDMENT));
+        return response;
+      })
+      .catch((err) => {
+        dispatch(error(reducerTypes.CREATE_PERMIT_AMENDMENT));
+        throw new Error(err);
+      })
+      .finally(() => dispatch(hideLoading("modal")));
+  };
+
+export const createPermitAmendmentVC =
+  (mineGuid: string, permitGuid: string, permitAmdendmentGuid: string) => (dispatch) => {
+    dispatch(request(reducerTypes.PERMIT_AMENDMENT_VC));
+    dispatch(showLoading());
+    return CustomAxios()
+      .post(
+        `${ENVIRONMENT.apiUrl}${API.PERMIT_AMENDMENT_VC(
+          mineGuid,
+          permitGuid,
+          permitAmdendmentGuid
+        )}`,
+        {},
+        createRequestHeader()
+      )
+      .then((response) => {
+        notification.success({
+          message: `Successfully Issued Permit Verifiable Credentials, Thanks Jason`,
+          duration: 10,
+        });
+        dispatch(success(reducerTypes.PERMIT_AMENDMENT_VC));
+        return response;
+      })
+      .catch((err) => {
+        dispatch(error(reducerTypes.PERMIT_AMENDMENT_VC));
+        throw new Error(err);
+      })
+      .finally(() => dispatch(hideLoading()));
+  };
+
+export const updatePermitAmendment =
+  (
+    mineGuid: string,
+    permitGuid: string,
+    permitAmdendmentGuid: string,
+    payload
```

Comment:
The payload type here is missing.

Reply:
Nice, thanks for the catch here. The interface for this is setup now.